### PR TITLE
Feature/atr 11 dev type typo in bql field name

### LINKED ISSUE #138: Create a check for DACs to valide fields
Could you include a verification when creating DAC or DACExtension to make sure each field has it's equivalent Property and vice-versa.  If one is missing then issue a warning.

public abstract class xxx : IBqlField { } // OK
public abstract class zzz : IBqlField { } // Issue warning

[PXDBInt]
public int? Xxx { get; set; } // OK

[PXDBInt]
public int? Yyy { get; set; } // Issue warning

### PR DESCRIPTION
**Pull request by SENya1990**
**MERGE AFTER PR https://github.com/Acumatica/Acuminator/pull/545**

Added diagnostic PX1066 for typos in BQL fields.

**Changes Overview:**

**PX1065 and PX1066**
- implemented PX1066 analysis in scope of PX1066 analyzer for missing BQL field
- reworked PX1065 analysis
- wrote unit tests for PX1066, corrected tests for PX1065
- made some refactoring and renaming
- wrote docs for PX1066

**Code Analysis Common Architecture and Infrastructure**
- introduced a base class `PXCodeFIxProvider` for code fix providers that contains common boilerplate logic for code action registration. Integrated new base type in most of code fix providers except the ones where custom logic was required. 
- Fixed incorrect calls to the `RegisterCodeFix` API - pass 1 diagnostic instead all context diagnostics.
- Added passing of cancellation tokens to the calls to Roslyn API 

**Semantic Model**
- Enhanced DAC field info with more flags
- Implemented `IEquatable<T>` by DAC field info and DAC BQL field info

**Other**
- refactoring, fixes of nullable analysis errors

### CODE REVIEW

File: docs/diagnostics/PX1066.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# PX1066
```
Comment (SENya1990) [inline docs/diagnostics/PX1066.md:1]: @EAndrosova please review new documentation for PX1066

File: docs/Summary.md [hunk 2]
```diff
@@ -72,6 +72,8 @@ Acuminator does not perform static analysis of projects whose names contain `Tes
 | [PX1062](diagnostics/PX1062.md) | You cannot declare static views, actions, mutable fields, and mutable properties in graphs or graph extensions  | Error | Available |
 | [PX1063](diagnostics/PX1063.md) | The declaration of the exception class does not contain a serialization constructor. This will cause an incorrect deserialization of the exception data and will lead to runtime errors in Acumatica ERP. | Error | Available |
 | [PX1064](diagnostics/PX1064.md) | The declaration of the exception class introduces new serializable fields but does not declare an override for the `GetObjectData` method. This will cause an incorrect serialization of the exception data and will lead to runtime errors in Acumatica ERP. | Error | Available |
+| [PX1065](diagnostics/PX1065.md) | The DAC field property does not have a corresponding DAC BQL field. | Error | Available |
+| [PX1066](diagnostics/PX1066.md) | There is probably a typo in the DAC BQL field name. | Warning (ISV Level 3: Informational) | Available |
```
Comment (SENya1990) [inline docs/Summary.md]: @EAndrosova please review the summary for PX1066

File: src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx [hunk 3]
```diff
@@ -402,4 +402,10 @@
   <data name="PX1022Graph" xml:space="preserve">
     <value>NonPublicGraph</value>
   </data>
+  <data name="PX1065" xml:space="preserve">
+    <value>NoBqlFieldForDacFieldProperty</value>
+  </data>
+  <data name="PX1066" xml:space="preserve">
+    <value>TypoInBqlFieldName</value>
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/DiagnosticsShortName.resx:409]: @EAndrosova please review the short name for PX1066

File: src/Acuminator/Acuminator.Analyzers/Resources.resx [hunk 4]
```diff
@@ -702,4 +702,16 @@ public virtual int? SomeDacField{ get; set; }</value>
   <data name="PX1022GraphExtensionFix" xml:space="preserve">
     <value>Make the graph extension public</value>
   </data>
+  <data name="PX1065TitleFormat" xml:space="preserve">
+    <value>The DAC field property "{0}" has Acumatica attributes on declared on it but does not have a corresponding BQL field. Such properties are not supported.</value>
+  </data>
+  <data name="PX1065FixFormat" xml:space="preserve">
+    <value>Add missing "{0}" BQL field to DAC</value>
+  </data>
+  <data name="PX1066TitleFormat" xml:space="preserve">
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/Resources.resx:711]: @EAndrosova please review diagnostic and code fix messages for PX1066

**Review by dropsonic — APPROVED** [review]

**Review by EAndrosova — APPROVED** [review]

**Pull request merged**